classes/class.tapestry-user-progress.php

```diff
@@ -216,6 +260,18 @@ private function _getUserH5PSettings()
         return $settings ? json_decode($settings) : (object) [];
     }
 
+    private function _updateAvatar($avatarData)
+    {
+        update_user_meta($this->_userId, 'avatar_'.$this->_userId, $avatarData);
+    }
+
+    private function _getAvatar()
+    {
+        $storedAvatar = get_user_meta($this->_userId, 'avatar_'.$this->_userId, true);
+
+        return $storedAvatar ? json_decode($storedAvatar) : (object) [];
+    }
```

Comment:
Since we are already saving this for this user, it's not needed to have their user ID in the key for this user meta. I also suggest adding something that makes it clear this is a tapestry user meta, e.g. `tapestry_avatar`